README.rst

```diff
@@ -51,6 +51,13 @@ It basically means:
 - generate metadata for haskell repo after main
   tree sync is done, using N+1 cores
 
+Build issues? If you see packages from the gentoo repository being
+selected over the haskell overlay packages and failing to build,
+check the priorities in the configuration files under
+``/etc/portage/repos.conf/``. The default priority for the gentoo
+overlay is -1000, so it should always be selected as last choice when
+version numbers are the same. 
+
```

Comment:
I don't consider it a "nitpic" - I am happy for input and guidance from others!  I also think this is a good idea, and will expand it to include details. I think I will also be adding to the genoo documentation, as the function of the priority isn't explained. In 18 years of using gentoo, I have never had reason to do anything with the priority!

Reply:
Looks good, although I'm not sure an example of `/etc/portage/repos.conf/gentoo.conf` is appropriate here since that is auto-generated and people won't need to edit it.

Also, I mentioned in #1271 that I think you would want to set the priority of `[haskell]` to something like `-1001`, e.g. a lower integer than `[gentoo]`, since it seems like lower values get higher priority. I may have to verify this, though.

Reply:
From my understanding, and based on my settings now, **higher values** get priority. I guess the intention is that anything you put on your system in overlays is probably what you want over the default.

Reply:
I will make the change you propose, but reference the gentoo.conf to see the current default as a reference to set the overlay priority.

Reply:
Ok, I checked it out and you are correct. Higher values get priority.

Since you ran into this because your `[gentoo]` entry didn't have the default priority value, it would actually make sense to leave it as you have it here.

Reply:
I should point out that my understanding is that when using the overlay, all the ebuilds should come from the overlay in order to avoid issues like the mtl issue I shared. Before changing the priority of the overlay, I noticed several packages scheduled to be merged which came from the gentoo repository rather than from the haskell overlay. I didn't get far enough to understand if these build or not.